airflow/www/security.py

```diff
@@ -235,6 +235,71 @@ def add_permissions(self, role, perms):
             perm_view = self.add_permission_view_menu(perm_name, view_name)
             self.add_permission_role(role, perm_view)
 
+    def get_resource(self, name):
+        """
+        Returns a resource record by name, if it exists.
+        :param name: Name of resource
+        :type name: str
+        :return: Resource record
+        :rtype: ViewMenu
+        """
+        return self.find_view_menu(name)
+
+    def get_all_resources(self):
+        """
+        Gets all existing resource records.
+        :return: List of all resources
+        :rtype: List[ViewMenu]
+        """
+        return self.get_all_view_menu()
+
+    def get_action(self, name):
+        """
+        Gets an existing action record.
+        :param name: name
+        :type name: str
+        :return: Action record, if it exists
+        :rtype: Permission
+        """
+        return super().find_permission(name)
```

Comment:
Why is this `super()` vs `self`?

Reply:
And should we change parameter to `action_name`?  for consistency ?

Reply:
Generally, I like to add only as much explanation as is necessary based on context. In this case it seems reasonable to infer that `name` refers to the name of the action being retrieved.